Do not use time zone conversion in datetimeoffset operations

staticlibs opened the pull request:
### Description

This change disables time zone conversion in `timestamp2tm` and `tm2timestamp` calls used in `datetimeoffset` operations.

### Issues Resolved

#1417

### Test Scenarios Covered ###

`TestDatetimeoffset`

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

staticlibs committed:
Do not use time zone conversion in datetimeoffset operations

Signed-off-by: Alex Kasko <alex@staticlibs.net>

2jungkook reviewed (changes requested):
The tests in Github action will succeed regardless of this change. In other words, this change can only be verified manually by running it in different timezone. Could you add a test case that can fail without this change? If needed, you can combine https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/1420 into this PR

staticlibs committed:
Test case added

Signed-off-by: Alex Kasko <alex@staticlibs.net>

staticlibs commented:
Added a test case, all the notes from [this 1412 comment](https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/1412#issuecomment-1529063874) also apply to this PR.

Despite being in the same area, this PR and #1412 are actually separate changes with different test failures. They cannot be both merged in current form because their code changes overlap in `datetimeoffset.c`. The intention is to have #1412 (that seems to be more serious) merged first and then to rebase  this PR on top of it and unify both test cases in the same `babel_datetimeoffset_timezone.mix` file.

staticlibs committed:
Rebase, test case update

Signed-off-by: Alex Kasko <alex@staticlibs.net>

staticlibs commented:
Rebased the PR on top of the latest `BABEL_3_X_DEV`.

Following up from [#1412](https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/1412#issuecomment-1536516143), updated `datetimeoffset-timezone-vu-prepare/verify/cleanup` and `datetimeoffset-timezone-before-15_3-vu-prepare/verify/cleanup` to change the time zone to `Australia/Sydney` before creating test table and view.

Removed `babel_datetimeoffset_timezone.mix` test, moved the check from it to `datetimeoffset-timezone-vu-verify`.

Changes to `tm2timestamp` calls in both `datetimeoffset_in` and in `datetimeoffset_pl_interval` do not change actual execution logic (thus there is no test coverage for them), because time zone conversion there was requested with `+00:00` time zone in all changed cases.

Changes to `timestamp2tm` in `datetimeoffset_pl_interval` are covered by the set of `dto + make_interval(...)` tests. Year, month, week and day intervals were actually affected by the change, time interval was added to test just in case.

Change to `timestamp2tm` in `datetimeoffset_date` is covered by the type casting check, in addition to `date` (the only type that had a problem)  all other date/time types are also covered.

Note, all checks specific to this fix were added only to `datetimeoffset-timezone-vu-verify`, NOT to `datetimeoffset-timezone-before-15_3-vu-verify`, I assume it is not necessary to add them to the latter because it is (being a `vu-verify` test) only run on the latest version.
